shopinvader_restapi/tests/common.py

```diff
@@ -0,0 +1,339 @@
+# Copyright 2017 Akretion (http://www.akretion.com).
+# @author Sébastien BEAU <sebastien.beau@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+# pylint: disable=method-required-super
+
+from contextlib import contextmanager
+from unittest import mock
+
+from odoo import fields
+from odoo.exceptions import MissingError
+from odoo.tests import TransactionCase
+
+from odoo.addons.base_rest.controllers.main import RestController
+from odoo.addons.base_rest.core import _rest_controllers_per_module
+from odoo.addons.base_rest.tests.common import BaseRestCase, RegistryMixin
+from odoo.addons.component.tests.common import ComponentMixin
+from odoo.addons.queue_job.job import Job
+from odoo.addons.shopinvader_restapi.models.track_external_mixin import (
+    TrackExternalMixin,
+)
+
+from .. import shopinvader_response, utils
+from ..services import abstract_download
+
+
+def _install_lang_odoo(env, lang_xml_id, full_install=False):
+    lang = env.ref(lang_xml_id)
+
+    # Full install can be very expensive as it reloads EVERY translation
+    # for EVERY installed module. 99.999% you don't need it for tests.
+    if full_install:
+        wizard = env["base.language.install"].create({"lang": lang.code})
+        wizard.lang_install()
+    else:
+        lang.active = True
+    return lang
+
+
+class UtilsMixin(object):
+    def _bind_products(self, products, backend=None):
+        backend = backend or self.backend
+        bind_wizard_model = self.env["shopinvader.variant.binding.wizard"]
+        bind_wizard = bind_wizard_model.create(
+            {
+                "backend_id": backend.id,
+                "product_ids": [(6, 0, products.ids)],
+                "run_immediately": True,
+            }
+        )
+        bind_wizard.bind_products()
+
+    def _refresh_json_data(self, products, backend=None):
+        """Force recomputation of JSON data for given products.
+
+        Especially helpful if your module adds new JSON keys
+        but the product are already there and computed w/out your key.
+        """
+        if not products:
+            return
+        backend = backend or self.backend
+        # TODO: remove hasattr check once `jsonify_stored` is ready.
+        # The json-store machinery comes from search engine module.
+        # We rely on it for product data BUT only
+        # `shopinvader_search_engine` requires that dependency.
+        # Hence, tests that need fresh product data because they add
+        # new keys to ir.exports record will be broken w/out refresh
+        # IF `shopinvader_search_engine` is installed (like on Travis).
+        # `jsonify_stored` will extrapolate the feature
+        # and allow to get rid of this hack.
+        # For full story see
+        # https://github.com/shopinvader/odoo-shopinvader/pull/783
+        if not hasattr(self.env["shopinvader.variant"], "recompute_json"):
```

Comment:
This condition has to be adapted as `shopinvader.variant` model doesn't exist anymore.